### DISCUSSION TITLE
Data types support object type

### DISCUSSION THREAD

**Discussion by liyan186** [GreptimeTeam/greptimedb#3593] [Ideas]
### What problem does the new feature solve?

Better store the following class

```
data class OrderBook(
    val symbol: String,
    val timeMillis: Long,
    val asks: List<Book>,
    val bids: List<Book>,
    val status: Boolean
) {
    data class Book(val price: BigDecimal, val size: BigDecimal)
}
```

### What does the feature do?

data types support object type. In this way, asks and bids  can store more efficiently.

**Comment by evenyag** [comment]
Thanks for your suggestion! I think we could implement the struct data type, both arrow and parquet support the struct data type.

**Comment by tisonkun** [comment]
Move to ideas until we have some concrete designs.
